File: common/src/main/java/com/datastrato/gravitino/dto/requests/SchemaUpdatesRequest.java
/*
 * Copyright 2023 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino.dto.requests;

import com.datastrato.gravitino.rest.RESTRequest;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

/** Represents a request to update a schema. */
@Getter
@EqualsAndHashCode
@ToString
public class SchemaUpdatesRequest implements RESTRequest {

  @JsonProperty("updates")
  private final List<SchemaUpdateRequest> updates;

  /**
   * Creates a new SchemaUpdatesRequest.
   *
   * @param updates The updates to apply to the schema.
   */
  public SchemaUpdatesRequest(List<SchemaUpdateRequest> updates) {
    this.updates = updates;
  }

  /** This is the constructor that is used by Jackson deserializer */
  public SchemaUpdatesRequest() {
    this(null);
  }

  /**
   * Validates the request.
   *
   * @throws IllegalArgumentException If the request is invalid, this exception is thrown.
   */
  @Override
  public void validate() throws IllegalArgumentException {
    updates.forEach(RESTRequest::validate);
  }
}

File: clients/client-python/gravitino/dto/requests/schema_updates_request.py
"""
Copyright 2024 Datastrato Pvt Ltd.
This software is licensed under the Apache License version 2.
"""

from dataclasses import dataclass, field
from typing import List

from dataclasses_json import config

from gravitino.dto.requests.schema_update_request import SchemaUpdateRequest
from gravitino.rest.rest_message import RESTRequest


@dataclass
class SchemaUpdatesRequest(RESTRequest):
    """Represents a request to update a schema."""

    _updates: List[SchemaUpdateRequest] = field(
        metadata=config(field_name="updates"), default_factory=list
    )

    def __init__(self, updates: List[SchemaUpdateRequest]):
        self._updates = updates

    def validate(self):
        """Validates the request.

        Raises:
            IllegalArgumentException If the request is invalid, this exception is thrown.
        """
        if not self._updates:
            raise ValueError("Updates cannot be empty")
        for update_request in self._updates:
            update_request.validate()
